src/vm/core.zig

```diff
@@ -282,37 +282,56 @@ pub const CairoVM = struct {
         self: *Self,
         instruction: *const instructions.Instruction,
     ) !OperandsResult {
-        // Compute the destination address and get value from the memory.
-        const dst_addr = try self.run_context.computeDstAddr(instruction);
-        const dst = try self.segments.memory.get(dst_addr);
-
-        // Compute the OP 0 address and get value from the memory.
-        const op_0_addr = try self.run_context.computeOp0Addr(instruction);
-        // Here we use `catch null` because we want op_0_op to be optional since it's not always used.
-        // TODO: identify if we need to use try or catch here.
-        const op_0_op = try self.segments.memory.get(op_0_addr);
-
-        // Compute the OP 1 address and get value from the memory.
-        const op_1_addr = try self.run_context.computeOp1Addr(
+
+        var op_res = OperandsResult.default();
+
+        op_res.dst_addr = try self.run_context.computeDstAddr(instruction);
+        op_res.dst = try self.segments.memory.get(op_res.dst_addr);
+        op_res.op_0_addr = try self.run_context.computeOp0Addr(instruction);
+        op_res.op_0 = try self.segments.memory.get(op_res.op_0_addr);
+
+        op_res.op_1_addr = try self.run_context.computeOp1Addr(
             instruction,
-            op_0_op,
+            op_res.op_0,
         );
-        const op_1_op = try self.segments.memory.get(op_1_addr);
 
-        const res = try computeRes(instruction, op_0_op, op_1_op);
+        op_res.op_1 = try self.segments.memory.get(op_res.op_1_addr);
+
+        op_res.res = try computeRes(instruction, op_res.op_0, op_res.op_1);
+        const res_op: ?MaybeRelocatable = op_res.res;
```

Comment:
I'm not sure but here you compute directly `computeRes` and you assign the result to the variable `res`. But for me what is done in Rust is a little different:
- The `res` variable is first initialized to `null`.
- `compute_op0_deductions` and `compute_op1_deductions` are calculated (if `op0_op` or `op1_op` are zero respectively, which is equivalent to an error for us), which can possibly modify the `res`.
- Then after these calculations, if `res` is still `null` then we calculate `compute_res` but not otherwise, which can lead to slightly different results I think.

Reply:
You are correct I jumbled up the moment of assigning `res`